src/coreclr/tools/aot/ILLink.Shared/DiagnosticId.cs

```diff
@@ -22,7 +22,7 @@ public enum DiagnosticId
 		CouldNotFindAssemblyReference = 1009,
 		CouldNotLoadAssembly = 1010,
 		FailedToWriteOutput = 1011,
-		LinkerUnexpectedError = 1012,
+		ILLinkUnexpectedError = 1012,
```

Comment:
This is a public API

Reply:
I feel like part of maintaining the DiagnosticId's is something that inevitably we have to do, there have been diagnosticId clashes in the past or ids that we no longer use and rename as `_unused_`. In fact, I was planning to, later on, fix some of the grammar problems in some of the DiagnosticId's (MissingArgumentForCommanLineOptionName = 1018, CannotConverValueToType = 1042, XmlUnkownBodyModification = 2011, XmlCouldNotFindGetAccesorOfPropertyOnType = 2018, XmlCouldNotFindSetAccesorOfPropertyOnType = 2019, XmlMoreThanOneValyForParameterOfMethod = 2024, XmlElementDoesNotContainRequiredAttributeFullname = 2029, XmlUnsuportedWildcard = 2100) which were flagged by the Visual Studio spellcheck feature.
I see this change as an improvement to readability which kind of falls in the same bucket for me, not so long ago we modify the SharedStrings to reflect reporting an illink issue in runtime when this Id is used.

Kind of unrelated I also noticed that this error id is used by dataflow, a piece that is reused by NativeAOT and although the message is not about opening an issue in runtime with the tag ILLink, I think we shouldn't be using this ID in other parts besides the Driver file in illink.

Reply:
I will see if I can rename it once I deal with the fact that Dataflow shouldn't produce this error id. For now I reverted the changes